controllers/machinehealthcheck_controller.go

```diff
@@ -0,0 +1,194 @@
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object - requeue the request.
+		logger.Error(err, "Failed to fetch MachineHealthCheck")
+		return ctrl.Result{}, err
+	}
+
+	cluster, err := util.GetClusterByName(ctx, r.Client, m.Namespace, m.Spec.ClusterName)
+	if err != nil {
+		logger.Error(err, "Failed to fetch Cluster for MachineHealthCheck")
+		return ctrl.Result{}, errors.Wrapf(err, "failed to get Cluster %q for MachineHealthCheck %q in namespace %q",
+			m.Spec.ClusterName, m.Name, m.Namespace)
+	}
+
+	// Return early if the object or Cluster is paused.
+	if util.IsPaused(cluster, m) {
+		logger.V(3).Info("reconciliation is paused for this object")
+		return ctrl.Result{}, nil
+	}
+
+	// Initialize the patch helper
+	patchHelper, err := patch.NewHelper(m, r.Client)
+	if err != nil {
+		logger.Error(err, "Failed to build patch helper")
+		return ctrl.Result{}, err
+	}
+
+	defer func() {
+		// Always attempt to patch the object and status after each reconciliation.
+		if err := patchHelper.Patch(ctx, m); err != nil {
+			reterr = kerrors.NewAggregate([]error{reterr, err})
+		}
+	}()
+
+	// Reconcile labels.
+	if m.Labels == nil {
+		m.Labels = make(map[string]string)
+	}
+	m.Labels[clusterv1.ClusterLabelName] = m.Spec.ClusterName
+
+	result, err := r.reconcile(ctx, cluster, m)
+	if err != nil {
+		logger.Error(err, "Failed to reconcile MachineHealthCheck")
+		r.recorder.Eventf(m, corev1.EventTypeWarning, "ReconcileError", "%v", err)
+
+		//TODO(JoelSpeed): Determine how/when to requeue requests if errors occur within r.reconcile
+	}
+
+	return result, nil
+}
+
+func (r *MachineHealthCheckReconciler) reconcile(_ context.Context, cluster *clusterv1.Cluster, m *clusterv1.MachineHealthCheck) (ctrl.Result, error) {
+	// Ensure the MachineHealthCheck is owned by the Cluster it belongs to
+	m.OwnerReferences = util.EnsureOwnerRef(m.OwnerReferences, metav1.OwnerReference{
+		APIVersion: clusterv1.GroupVersion.String(),
+		Kind:       "Cluster",
+		Name:       cluster.Name,
+		UID:        cluster.UID,
+	})
+
+	return ctrl.Result{}, fmt.Errorf("controller not yet implemented")
+}
+
+func (r *MachineHealthCheckReconciler) indexMachineHealthCheckByClusterName(object runtime.Object) []string {
+	mhc, ok := object.(*clusterv1.MachineHealthCheck)
+	if !ok {
+		r.Log.Error(errors.New("incorrect type"), "expected a MachineHealthCheck", "type", fmt.Sprintf("%T", object))
+		return nil
+	}
+
+	return []string{mhc.Spec.ClusterName}
+}
+
+// clusterToMachineHealthCheck maps events from Cluster objects to
+// MachineHealthCheck objects that belong to the Cluster
+func (r *MachineHealthCheckReconciler) clusterToMachineHealthCheck(o handler.MapObject) []reconcile.Request {
+	c, ok := o.Object.(*clusterv1.Cluster)
+	if !ok {
+		r.Log.Error(errors.New("incorrect type"), "expected a MachineHealthCheck", "type", fmt.Sprintf("%T", o))
```

Comment:
@ncdc Sorry, I just realised I've made a typo here, will fix-up and re-push
```suggestion
		r.Log.Error(errors.New("incorrect type"), "expected a Cluster", "type", fmt.Sprintf("%T", o))
```